src/TypeScript/EslintBridgeClient/EslintBridgeStartUp.cs

```diff
@@ -19,17 +19,183 @@
 
 namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient
 {
-    // TODO: https://github.com/SonarSource/sonarlint-visualstudio/issues/2256
-
     internal interface IEslintBridgeStartUp : IDisposable
     {
         /// <summary>
         /// Ensures that the process is running. Returns the port that server is running on.
         /// </summary>
         Task<int> Start();
     }
+
+    [Export(typeof(IEslintBridgeStartUp))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class EslintBridgeStartUp : IEslintBridgeStartUp
+    {
+        internal const string EslintBridgeDirectoryMefContractName = "SonarLint.TypeScript.EsLintBridgeServerPath";
+        
+        private readonly string eslintBridgeStartupScriptPath;
+        private readonly INodeLocator nodeLocator;
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+        private TaskCompletionSource<int> startTask;
+
+        internal Process Process;
+
+        [ImportingConstructor]
+        public EslintBridgeStartUp([Import(EslintBridgeDirectoryMefContractName)] string eslintBridgeStartupScriptPath,
+            INodeLocator nodeLocator,
+            ILogger logger)
+            : this(eslintBridgeStartupScriptPath, nodeLocator, new FileSystem(), logger)
+        {
+        }
+
+        internal EslintBridgeStartUp(string eslintBridgeStartupScriptPath,
+            INodeLocator nodeLocator,
+            IFileSystem fileSystem,
+            ILogger logger)
+        {
+            this.eslintBridgeStartupScriptPath = eslintBridgeStartupScriptPath;
+            this.nodeLocator = nodeLocator;
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
+        public Task<int> Start()
+        {
+            if (startTask != null && !startTask.Task.IsFaulted)
+            {
+                return startTask.Task;
+            }
+
+            startTask = new TaskCompletionSource<int>();
+            try
+            {
+                StartServer();
+            }
+            catch (Exception ex)
+            {
+                startTask.SetException(ex);
+            }
+
+            return startTask.Task;
+        }
+
+        public void Dispose()
+        {
+            Stop();
+        }
+
+        private void StartServer()
+        {
+            var nodePath = FindNodePath();
+            var serverStartupScriptLocation = FindServerScriptFile();
+
+            var psi = new ProcessStartInfo
+            {
+                FileName = nodePath,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false, // required if we want to capture the error output
+                ErrorDialog = false,
+                CreateNoWindow = true,
+                Arguments = serverStartupScriptLocation
+            };
+
+            psi.EnvironmentVariables.Add("NODE_PATH", @"%APPDATA%\npm\node_modules");
```

Comment:
I'm having a trouble to test this. I thought about creating a file that outputs this to another file but that's just over-complicated.
An alternative is to inherit from Process and wrap `Process.Start`, but I'm not sure that it's worth it.

Reply:
Not sure; it might not be necessary. There was a lot of temporary/experimental stuff in the prototype, including some bits that were ported from the Java code.

I think we should drop it, and add it only if it turns out to be necessary.